sympy/polys/matrices/tests/test_domainscalar.py

```diff
@@ -0,0 +1,82 @@
+from sympy.testing.pytest import raises
+
+from sympy.polys import ZZ, QQ
+from sympy.polys.matrices.domainscalar import DomainScalar
+
+
+def test_DomainScalar_init():
+    A = DomainScalar(ZZ(1), ZZ)
+    assert A.element ==  ZZ(1)
+    assert A.domain == ZZ
+
+
+def test_DomainScalar_new():
+    A = DomainScalar(ZZ(1), ZZ)
+    B = A.new(ZZ(4), ZZ)
+    assert B == DomainScalar(ZZ(4), ZZ)
+
+
+def test_DomainScalar_repr():
+    A = DomainScalar(ZZ(1), ZZ)
+    assert A.__repr__() == '1'
+
+
+def test_DomainScalar_from_sympy():
+    expr = ZZ(1)
+    B = DomainScalar.from_sympy(expr)
+    assert B == DomainScalar(ZZ(1), ZZ)
+
+
+def test_DomainScalar_to_domain():
+    A = DomainScalar(ZZ(1), ZZ)
+    B = A.to_domain(QQ)
+    assert B == DomainScalar(QQ(1), QQ)
+
+
+def test_DomainScalar_unify():
+    A = DomainScalar(ZZ(1), ZZ)
+    B = DomainScalar(QQ(2), QQ)
+    A, B = A.unify(B)
+    assert A.domain == B.domain == QQ
+
+
+def test_DomainScalar_add():
+    A = DomainScalar(ZZ(1), ZZ)
+    B = DomainScalar(QQ(2), QQ)
+    C = A.__add__(B)
+    assert C == DomainScalar(QQ(3), QQ)
+
+
+def test_DomainScalar_sub():
+    A = DomainScalar(ZZ(1), ZZ)
+    B = DomainScalar(QQ(2), QQ)
+    C = A.__sub__(B)
+    assert C == DomainScalar(QQ(-1), QQ)
```

Comment:
This should test `C = A - B` not calling `__sub__` explicitly.

Reply:
Sure, will change for all the relevant methods as well.
Thanks.